File: frontend/types/project.ts
export type ProjectType = "new_building" | "existing_building";

export type ProjectStatus =
  | "draft"
  | "documents_uploaded"
  | "processing"
  | "analyzed"
  | "archived";

export interface Project {
  id: string;
  organization_id: string | null;
  created_by: string;

  name: string;
  description: string | null;

  city: string | null;
  district: string | null;

  building_type: string | null;
  project_type: ProjectType;

  total_construction_area: number | null;
  parcel_area: number | null;
  floor_count: number | null;
  main_facade_direction: string | null;

  target_certificate_level: string | null;
  estimated_budget: number | null;

  status: ProjectStatus;

  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/project.py
from decimal import Decimal
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


ProjectType = Literal["new_building", "existing_building"]


class ProjectCreate(BaseModel):
    name: str = Field(min_length=3, max_length=200)
    description: str | None = None

    city: str | None = Field(default=None, max_length=100)
    district: str | None = Field(default=None, max_length=100)

    building_type: str | None = Field(default=None, max_length=100)
    project_type: ProjectType = "new_building"

    total_construction_area: Decimal | None = Field(default=None, gt=0)
    parcel_area: Decimal | None = Field(default=None, gt=0)
    floor_count: int | None = Field(default=None, gt=0)

    main_facade_direction: str | None = None
    target_certificate_level: str | None = None
    estimated_budget: Decimal | None = Field(default=None, gt=0)


class ProjectResponse(ProjectCreate):
    id: UUID
    created_by: UUID
    status: str

    model_config = {
        "from_attributes": True
    }